### ISSUE TITLE
Adapter + LLM System

### ISSUE THREAD

**Issue by YES-DSP** [YES-DSP/DSP_LAB#7]
Adapter를 이용해서 서로 다른 도메인을 가진 데이터를 LLM 지식모델에 연결할 수 있다.
이러한 개념을 통해서 여러 도메인의 정보를 조합하는 Recommander System을 구축하는 것을 목적으로 한다.

**Comment by YES-DSP** [comment]
# Using Practice Data

Dacon 인공지능 경진대회에서 종료된 대회의 데이터와 평가 알고리즘을 사용하였다. 이를 통해서 도메인의 연결이 추천 알고리즘의 성능을 개선할 수 있는지를 확인하고자 한다. 해당 대회는 24년 7월에 종료된 대회로, 웹기사 추천 알고리즘을 설계하는 것을 목적으로 한다. 각각 유저가 웹기사를 조회한 이력에 대한 데이터, 웹기사의 데이터를 이용해서 최적 알고리즘을 설계하는 것을 목적으로 한다. 

# Using Matrix Factorization

우선 비교군을 위해서 가장 기본적인 추천 시스템 알고리즘인 Matrix Factorization을 이용해서 데이터를 분석하였다.
`import pandas as pd
import numpy as np

from sklearn.metrics.pairwise import cosine_similarity

view_log_train = pd.read_csv('data/view_log.csv')
article_info = pd.read_csv('data/article_info.csv')
submission = pd.read_csv('data/sample_submission.csv')

user_article_matrix = view_log_train.groupby(['userID', 'articleID']).size().unstack(fill_value=0)
print(user_article_matrix)
user_article_matrix.to_csv('user_article_matrix.csv', index=True)

user_similarity = cosine_similarity(user_article_matrix)

user_similarity

user_predicted_scores = user_similarity.dot(user_article_matrix) / np.array([np.abs(user_similarity).sum(axis=1)]).T

recommendations = []
for idx, user in enumerate(user_article_matrix.index):    
    sorted_indices = user_predicted_scores[idx].argsort()[::-1]
    top5recommend = [article for article in user_article_matrix.columns[sorted_indices]][:5]
    
    for article in top5recommend:
        recommendations.append([user, article])
        
top_recommendations = pd.DataFrame(recommendations, columns=['userID', 'articleID'])

submission['articleID'] = top_recommendations['articleID']

submission.to_csv('baseline_submission.csv', index=False)
`

**Comment by bowonlee-inha** [comment]
<img width="485" alt="Image" src="https://github.com/user-attachments/assets/2f0c056f-d2af-4dfa-8076-b4baa98fd83b" />
